api/src/main/java/org/openmrs/module/coreapps/customdatatype/CodedConceptDatatype.java

```diff
@@ -0,0 +1,29 @@
+package org.openmrs.module.coreapps.customdatatype;
+
+import org.openmrs.customdatatype.SerializingCustomDatatype;
+import org.springframework.stereotype.Component;
+
+/**
+ * This class allows for coded concepts to be displayed as a drop-down when creating visit attributes.
+ * The concept id MUST be set in the datatype configuration field.
+ */
+@Component
+public class CodedConceptDatatype extends SerializingCustomDatatype<String> {
+
+	/**
+	 * @see org.openmrs.customdatatype.SerializingCustomDatatype#serialize(java.lang.Object)
+	 */
+	@Override
+	public String serialize(String typedValue) {
+		return typedValue;
+	}
+
+	/**
+	 * @see org.openmrs.customdatatype.SerializingCustomDatatype#deserialize(java.lang.String)
+	 */
+	@Override
+	public String deserialize(String serializedValue) {
```

Comment:
These are the datatypes which appear as combo box selections on the visit attributes page. Notice ConceptDatatype extends BaseOpenmrsMetadata whereas CodedConceptDatatype extends SerializingCustomDatatype. Look at the DateDatatype, BooleanDatetype, Freetextdatatype for similar functionality.